src/dex/wombat/wombat.ts

```diff
@@ -0,0 +1,436 @@
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    protected adapters = Adapters[network] || {}, // TODO: add any additional optional params to support other fork DEXes
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.config = WombatConfig[dexKey][network];
+    this.pollingManager = StatePollingManager.getInstance(dexHelper);
+
+    this.bmw = new WombatBmw(
+      dexKey,
+      this.config.bmwAddress,
+      network,
+      dexHelper,
+      this.logger,
+      this.config.bmwAddress,
+      this.onAssetAdded.bind(this),
+    );
+    this.pollingManager.initializeAllPendingPools();
+  }
+
+  async init(blockNumber: number) {
+    if (!this.bmw.isInitialized) {
+      await this.bmw.initialize(blockNumber);
+    }
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {
+    await this.init(blockNumber);
+    const bmwState = this.bmw.getState(blockNumber);
+    if (!bmwState) {
+      throw new Error('initializePricing: bmwState still null after init');
+    }
+  }
+
+  onAssetAdded = async (
+    pool: Address,
+    asset2TokenMap: Map<Address, Address>,
+    blockNumber: number,
+  ): Promise<void> => {
+    if (!this.pools[pool]) {
+      this.pools[pool] = new WombatPool(
+        this.dexKey,
+        this.getPoolIdentifier(pool),
+        this.dexHelper,
+        pool,
+        asset2TokenMap,
+      );
+      await this.pools[pool].getState('latest', true);
+    } else {
+      this.pools[pool].addAssets(asset2TokenMap);
+    }
+  };
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return this.adapters[side] ? this.adapters[side] : null;
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifiers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    await this.updatePoolState();
+    return (
+      await this.findPools(
+        this.dexHelper.config.wrapETH(srcToken).address.toLowerCase(),
+        this.dexHelper.config.wrapETH(destToken).address.toLowerCase(),
+        blockNumber,
+      )
+    ).map(p => this.getPoolIdentifier(p));
+  }
+
+  protected getPoolIdentifier(poolAddress: Address): string {
+    return `${this.dexKey}_${poolAddress}`;
+  }
+
+  // Returns pool prices for amounts.
+  // If limitPools is defined only pools in limitPools
+  // should be used. If limitPools is undefined then
+  // any pools can be used.
+  async getPricesVolume(
```

Comment:
@KanievskyiDanylo As far as I can see we don't support buy for Wombat but I don't see condition like `if (side === SwapSide.BUY) return null;` in `getPricesVolume` method and in other places. Could you please use [Solidly](https://github.com/paraswap/paraswap-dex-lib/blob/master/src/dex/solidly/solidly.ts#L301) as a reference dex which don't support buy too and fix this?

Reply:
@KanievskyiDanylo 

<img width="1109" alt="paraswap-dex-lib:src:dex:solidly:solidly ts at master · paraswap:paraswap-dex-lib 🔊 2023-12-08 13-50-28" src="https://github.com/paraswap/paraswap-dex-lib/assets/2093810/d505903c-ca88-4ad8-8412-a004b23a7113">

<img width="1115" alt="paraswap-dex-lib:src:dex:solidly:solidly ts at master · paraswap:paraswap-dex-lib 🔊 2023-12-08 13-52-06" src="https://github.com/paraswap/paraswap-dex-lib/assets/2093810/fd30923c-a481-46e7-bea0-b0362a82365a">
<img width="1111" alt="paraswap-dex-lib:src:dex:solidly:solidly ts at master · paraswap:paraswap-dex-lib 🔊 2023-12-08 13-51-52" src="https://github.com/paraswap/paraswap-dex-lib/assets/2093810/82d38583-413e-41f1-91be-61d5959e71de">